cluster_utils/job.py

```diff
@@ -189,10 +193,10 @@ def generate_execution_cmd(self, paths, cmd_prefix: Optional[str] = None):
         res = "\n".join(
             [
                 set_cwd,
+                set_env_variables,
                 pre_job_script,
                 virtual_env_activate,
                 conda_env_activate,
-                set_env_variables,
                 exec_cmd,
             ]
```

Comment:
The documentation currently explicitly says
> Variables are set after a virtual/conda environment is activated, thus override environment variables set before.

Would it make sense to move `virtual_env_activate` and `conda_env_activate` up as well to preserve this behaviour?  This would of course affect the environment of the pre_job_script, which might be desirable or not, depending on how it is used.

See https://github.com/martius-lab/cluster_utils/blob/0568241666815ed2252654d65e75b6658d1eae58/docs/configuration.rst?plain=1#L98-L100

Reply:
Not sure what's the best way to handle this, probably does not matter very much.